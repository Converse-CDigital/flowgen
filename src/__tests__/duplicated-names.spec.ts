import { compiler, beautify } from "..";
import "../test-matchers";

it("should handle variable & type having same name", () => {
  const ts = `
  export declare const AuthMechanism: {
    readonly MONGODB_AWS: "MONGODB-AWS";
    readonly MONGODB_CR: "MONGODB-CR";
  };
  export declare type AuthMechanism = typeof AuthMechanism[keyof typeof AuthMechanism];
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});

it("should generate unique names", () => {
  const ts = `
  export declare type AuthMechanismType = string;
  export declare const AuthMechanism: {
    readonly MONGODB_AWS: "MONGODB-AWS";
    readonly MONGODB_CR: "MONGODB-CR";
  };
  export declare type AuthMechanism = typeof AuthMechanism[keyof typeof AuthMechanism];
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
  expect(result).toBeValidFlowTypeDeclarations();
});

it("should be scoped to main file", () => {
  const ts = `
  import { Buffer } from 'buffer';
  export declare type BufferAlias = Buffer;
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
  // FlowJs is not valid due to `import` not transpiled to typeof
  // expect(result).toBeValidFlowTypeDeclarations();
});
