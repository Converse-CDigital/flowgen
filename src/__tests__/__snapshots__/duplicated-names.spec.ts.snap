// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be scoped to main file 1`] = `
"import { Buffer } from \\"buffer\\";
export type BufferAlias = Buffer;
"
`;

exports[`should generate unique names 1`] = `
"export type AuthMechanismType = string;
declare export var AuthMechanism: {
  +MONGODB_AWS: \\"MONGODB-AWS\\",
  +MONGODB_CR: \\"MONGODB-CR\\",
  ...
};
export type AuthMechanismType1 = $ElementType<
  typeof AuthMechanism,
  $Keys<typeof AuthMechanism>
>;
"
`;

exports[`should handle variable & type having same name 1`] = `
"declare export var AuthMechanism: {
  +MONGODB_AWS: \\"MONGODB-AWS\\",
  +MONGODB_CR: \\"MONGODB-CR\\",
  ...
};
export type AuthMechanismType = $ElementType<
  typeof AuthMechanism,
  $Keys<typeof AuthMechanism>
>;
"
`;
